Update F# optional parameters to expound upone default parameter values (#10586)
dotnet/docs: docs/fsharp/language-reference/parameters-and-arguments.md

        printfn "%s" message
```

> You can also specify new objects as a default parameter value. For example, the `Foo` member could have an optional `CanceallationToken` as input instead:

Reviewer: a new object as a default parameter value? The plural the singular don't go well together.

```fsharp
open System.Threading
open System.Runtime.InteropServices
type C = 
    static member Foo([<Optional; DefaultParameterValue(CancellationToken())>] ct: CancellationToken) =
        printfn "%A" ct
```

> The value given as argument to `DefaultParameterValue` must match the type of the parameter, i.e. the following is not allowed:

Reviewer: i.e. shouldn't be used for localization reasons.
```suggestion
The value given as argument to `DefaultParameterValue` must match the type of the parameter. For example, the following is not allowed:
```
